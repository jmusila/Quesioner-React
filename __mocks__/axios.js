module.exports = {
    get: () => {
      return Promise.resolve({
        data: [
          {
            id: 0,
            name: "go home now"
          },
          {
            id: 1,
            name: "do something at Andela"
          }
        ]
      });
    }
  };