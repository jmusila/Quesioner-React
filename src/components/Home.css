* {
    box-sizing: border-box;
}
body{
    font-family: "Oxygen", "Ubuntu", "Roboto";
    padding-top: 0em;
}
hr{
    width: 35em;
}

.nav-header {
    color: #0099CC;
    margin-left: 2em;
    float: left;
    font-size: 35px;
    padding: 30px 0px;
}
.card {
    margin: 0.4rem;
    border: solid 1px rgba(0, 0, 0, 0.1);
    border-radius: 5px;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    width: 100%;
    position: relative;
    padding: 10px;
    display: flex;
    flex-direction: column;
}