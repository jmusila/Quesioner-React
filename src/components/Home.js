import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import "./Home.css";
import { Container, Col, Row } from 'reactstrap';
import { fetchActions } from '../actions/homeActions';

class Home extends Component {

  componentWillMount() {
    this.props.fetchActions();
  }
  render() {
    let meetups = this.props.meetups.map(meetup => {
      return (
        <Container>
          <div className="card">
            <h2><b>Title:</b> {meetup.title}</h2>
            <img src="https://images.unsplash.com/photo-1543269865-cbf427effbad?ixlib=rb-1.2.1&auto=format&fit=crop&w=1950&q=80" width="100%" height="350" />
            <p><b>Location:</b> {meetup.location}</p>
            <p><b>Date Happening:</b> {meetup.happeningOn}</p>
            <p><b>Tags:</b> {meetup.tags}</p>
            <p><b>Time Added</b> {meetup.time_added}</p>
          </div>
        </Container>

      );
    });
    return (
      <Col sm="12"><Row>{meetups}</Row></Col>
    );
  }
}

Home.propTypes = {
  fetchActions: PropTypes.func.isRequired,
  meetups: PropTypes.array.isRequired,
}

const mapStateToProps = state => ({
  meetups: state.meetups.data,
});
export default connect(mapStateToProps, { fetchActions })(Home);

