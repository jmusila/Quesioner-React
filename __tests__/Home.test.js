import React from "react";
import { shallow } from "enzyme";
import axios from "axios";
import configureMockStore from 'redux-mock-store';
import Home from "../src/components/Home";

jest.mock("axios");
const mockStore = configureMockStore();

describe('Test home Component', () => {
  let store;
  let wrapper;
  const getSpy = jest.spyOn(axios, "get");

  beforeEach(() => {
    store = mockStore({});
    wrapper = shallow(<Home store={store} />);
  });
  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("renders with correct state", () => {
    expect(wrapper.state("meetups")).toEqual([]);
  });
  test("mock fetch list of meetups from api", () => {
    expect(getSpy).toBeCalled();
    getSpy.mockClear();
  });
})
