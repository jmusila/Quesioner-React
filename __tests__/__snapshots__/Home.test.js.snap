// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Test home Component should render without crashing 1`] = `undefined`;
